### PR TITLE
Make formatter stable

### PR DESCRIPTION
**Pull request by stackoverflow**

### CODE REVIEW

File: pkl-formatter/src/main/kotlin/org/pkl/formatter/Builder.kt [hunk 1]
```diff
@@ -621,7 +621,7 @@ internal class Builder(sourceText: String, private val grammarVersion: GrammarVe
     return if (node.children.isNotEmpty()) {
       endsWithClosingBracket(node.children.last())
     } else {
-      node.isTerminal("}")
+      node.isTerminal("}") || node.type.isAffix
```
Comment (stackoverflow) [inline pkl-formatter/src/main/kotlin/org/pkl/formatter/Builder.kt:624]: This has the side effect of not adding a newline even if the last element is a single line block comment:
```pkl
items = allItems.filter((it) ->
            it /*foo*/
          )

// formats to
items =
  allItems.filter((it) -> it
  /*foo*/)
```

But this is such an edge case I don't think it worth writing special handling.

File: pkl-formatter/src/test/files/FormatterSnippetTests/output/line-breaks.pkl [hunk 2]
```diff
@@ -74,11 +74,11 @@ local function render(currentIndent: String) =
...
-//     badItems.containsKey(item) // some line comment 
-//     || item.tags.toList().findOrNull((it) -> it.type == "bookmark") != null // some other line comment
-//   )
-//     .sortBy((item) -> item.name)
+items: List<Item> =
+  allItems
+    .filter((item) ->
+      badItems.containsKey(item) // some line comment 
+        || item.tags.toList().findOrNull((it) -> it.type == "bookmark") != null
+    // some other line comment
```
Comment (HT154) [inline pkl-formatter/src/test/files/FormatterSnippetTests/output/line-breaks.pkl]: Is this really correct? It seems odd to move a line comment off its original line.

Comment (stackoverflow) [reply pkl-formatter/src/test/files/FormatterSnippetTests/output/line-breaks.pkl]: Indeed. It shouldn't be on a newline. Probably a side effect of the trailing lambda rule. Will take a look.

Comment (stackoverflow) [reply pkl-formatter/src/test/files/FormatterSnippetTests/output/line-breaks.pkl]: Fixed now.
